src/Infrastructure/Service/WorkflowTransitionResolverRegistry.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Service;
+
+use SprykerSdk\SdkContracts\Workflow\TransitionResolverInterface;
+
+class WorkflowTransitionResolverRegistry
+{
+    /**
+     * @var iterable<\SprykerSdk\SdkContracts\Workflow\TransitionResolverInterface>
+     */
+    protected iterable $workflowTransitionResolvers;
+
+    /**
+     * @var array<string, \SprykerSdk\SdkContracts\Workflow\TransitionResolverInterface>|null
+     */
+    protected ?array $transitionResolverMap = null;
```

Comment:
Does it make sense to have nullable array here?
Let's just put `protected array $transitionResolverMap = [];`

Reply:
How can i check the registry has been populated already?